plugin/src/main/groovy/com/novoda/staticanalysis/StaticAnalysisPlugin.groovy

```diff
@@ -10,39 +11,45 @@ import com.novoda.staticanalysis.internal.pmd.PmdConfigurator
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.Plugin
 import org.gradle.api.Project
-import org.gradle.api.Task
+
+import static com.novoda.staticanalysis.internal.TasksCompat.createTask
 
 class StaticAnalysisPlugin implements Plugin<Project> {
 
     @Override
     void apply(Project project) {
         def pluginExtension = project.extensions.create('staticAnalysis', StaticAnalysisExtension, project)
         def evaluateViolations = createEvaluateViolationsTask(project, pluginExtension)
-        createConfigurators(project, pluginExtension, evaluateViolations).each { configurator -> configurator.execute() }
+        createConfigurators(project, pluginExtension).each { configurator -> configurator.execute() }
         project.afterEvaluate {
             project.tasks['check'].dependsOn evaluateViolations
         }
+
     }
 
-    private static Task createEvaluateViolationsTask(Project project,
-                                                     StaticAnalysisExtension extension) {
-        project.tasks.create('evaluateViolations', EvaluateViolationsTask) { task ->
+    private static def createEvaluateViolationsTask(Project project, StaticAnalysisExtension extension) {
+        createTask(project, 'evaluateViolations', EvaluateViolationsTask) { task ->
             task.evaluator = { extension.evaluator }
             task.allViolations = { extension.allViolations }
+
+            project.tasks.withType(ToolTriggerTask) { toolTrigger ->
+                task.dependsOn(toolTrigger)
+            }
+            project.tasks.withType(CollectViolationsTask) { collectViolations ->
```

Comment:
`dependsOn` relation is now automatically added for all types of `CollectViolationsTask`s. No need to pass `evaluateViolations` anymore. 

But I have to put it into configuration of `evaluateViolations` which means this relation is lazily put once `evaluateViolations` is run.